### PR TITLE
Throw a JsonMapperException if php configuration discards comments.

### LINKED ISSUE #13: Throw a configuration exception if comments are disabled
We know that our JsonMapper is dependent upon annotations in method's comments, so we need to make sure that comments are enabled.
There is a configuration `opcache.save_comments` that can disable comments and cause issues while deserialization.

We need to throw a configuration exception if `opcache.save_comments` or any other similar configuration is set to off as done by [Doctrine's AnnotationException checks](https://github.com/doctrine/annotations/blob/39f2a48113777ce9f5e02c04c5c1ba4ed80b9c3d/lib/Doctrine/Common/Annotations/AnnotationReader.php#L111-L120)

### PR DESCRIPTION
**Pull request by Mohammad-Haris**
Closes #13

### CODE REVIEW

File: src/JsonMapperException.php [hunk 1]
```diff
@@ -24,5 +24,24 @@
...
 class JsonMapperException extends \Exception
 {
+    /**
+     * Exception for discarded comments setting in configuration.
+     * 
+     * @param array $concernedKeys Keys (PHP direcitves) with issues
+     * 
+     * @return JsonMapperException
+     */
+    static function commentsDisabledInConfigurationException($concernedKeys)
```
Comment (asadali214) [inline src/JsonMapperException.php:34]: Very good approach to define exception messages in JsonMapperException class

**Comment by asadali214** [comment]
This change looks good but we need to add its unit test that test this Exception being thrown and compare the exception message

checkout unitTest: JsonMapperTest::testMapClassWithNonObject() to see how to expect an exception and its message

**Review by asadali214 — CHANGES_REQUESTED** [review]
Kindly add the unit test to test this change

File: src/JsonMapperException.php (hunk 1 above)
Comment (Mohammad-Haris) [reply src/JsonMapperException.php:34]: Not an original idea, saw this approach being used in the annotations repo you linked in the bug.
I probably would have gone with a function, but not exactly. The static function creating the object internally is good.

File: src/JsonMapper.php [hunk 2]
```diff
@@ -93,6 +93,36 @@ class JsonMapper
      */
     protected $arInspectedClasses = array();
 
+    /**
+     * Constructor for JsonMapper.
+     */
+    function __construct()
```
Comment (asadali214) [inline src/JsonMapper.php:99]: This constructor is throwing an Exception, please add the throws tag for it

**Review by asadali214 — APPROVED** [review]
These changes looks good to me
Good work done @Mohammad-Haris

**Pull request merged**